src/Diagnostics.ModelsAndUtils/Attributes/SupportTopic.cs

```diff
@@ -14,14 +14,24 @@ public class SupportTopic : Attribute, IEquatable<SupportTopic>
         /// See <see href="http://aka.ms/selfhelppreview"/>
         public string Id;
 
+        /// <summary>
+        /// Sap Support Topic Id
+        /// </summary>
+        public string SapSupportTopicId;
+
+        /// <summary>
+        /// Sap product Id
+        /// </summary>
+        public string SapProductId;
+
         /// <summary>
         /// Unique resource Id.
         /// </summary>
         public string PesId;
 
         public bool Equals(SupportTopic other)
         {
-            return (this.Id == other.Id && this.PesId == other.PesId);
+            return (this.Id == other.Id && this.PesId == other.PesId || this.SapSupportTopicId == other.SapSupportTopicId && this.SapProductId == other.SapProductId);
```

Comment:
this should be :

((this.Id == other.Id && this.PesId == other.PesId) || (this.SapSupportTopicId == other.SapSupportTopicId && this.SapProductId == other.SapProductId))

Reply:
"&&" has higher priority than "||" right Shekhar? So I guess they work the same?
For code readability , what you suggested is definitely better to understand.